shell/platform/linux/public/flutter_linux/fl_texture_registrar.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include <stdint.h>
+
+#include "fl_texture.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlTextureRegistrar,
+                     fl_texture_registrar,
+                     FL,
+                     TEXTURE_REGISTRAR,
+                     GObject)
+
+/**
+ * FlTextureRegistrar:
+ *
+ * #FlTextureRegistrar is used when registering textures.
+ *
+ * Flutter Framework accesses your texture by the related unique texture ID. To
+ * draw your texture in Dart, you should add Texture widget in your widget tree
+ * with the same texture ID. Use platform channels to send this unique texture
+ * ID to the Dart side.
+ */
+
+/**
+ * fl_texture_registrar_register_texture:
+ * @registrar: an #FlTextureRegistrar.
+ * @texture: an #FlTexture for registration.
+ *
+ * Registers a texture and returns the unique ID for that texture.
+ *
+ * Returns: unique ID of the texture.
+ */
+int64_t fl_texture_registrar_register_texture(FlTextureRegistrar* registrar,
```

Comment:
The ID is always the same as the memory address of `texture`, so it is an unnecessary concept in the API. You can make it a static function inside `fl_texture_registrar.cc` and the following two functions can then just take `FlTexture*` as an argument. This is much simpler.